numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,19 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=np._NoValue, *args, **kwargs):
```

Comment:
I would just use `None` here rather than `np._NoValue`. It's consistent with the docstring, and `None` isn't a reasonable value for the omitted argument (which should be callable as a function).